crates/api_crud/src/site/create.rs

```diff
@@ -41,32 +44,13 @@ impl PerformCrud for CreateSite {
   #[tracing::instrument(skip(context))]
   async fn perform(&self, context: &Data<LemmyContext>) -> Result<SiteResponse, LemmyError> {
     let data: &CreateSite = self;
-
-    let local_site = LocalSite::read(context.pool()).await?;
-
-    if local_site.site_setup {
-      return Err(LemmyError::from_message("site_already_exists"));
-    };
-
     let local_user_view = local_user_view_from_jwt(&data.auth, context).await?;
+    let local_site = LocalSite::read(context.pool()).await?;
 
-    let sidebar = diesel_option_overwrite(&data.sidebar);
-    let description = diesel_option_overwrite(&data.description);
-    let icon = diesel_option_overwrite_to_url(&data.icon)?;
-    let banner = diesel_option_overwrite_to_url(&data.banner)?;
-
-    let slur_regex = local_site_to_slur_regex(&local_site);
-    check_slurs(&data.name, &slur_regex)?;
-    check_slurs_opt(&data.description, &slur_regex)?;
-
-    // Make sure user is an admin
+    // Make sure user is an admin; other types of users should not create site data...
     is_admin(&local_user_view)?;
 
-    if let Some(Some(desc)) = &description {
-      site_description_length_check(desc)?;
-    }
-
-    is_valid_body_field(&data.sidebar)?;
+    validate_create_payload(local_site.site_setup, local_site.slur_filter_regex, data)?;
 
     let application_question = diesel_option_overwrite(&data.application_question);
```

Comment:
I want to move this into the validation fn in the next PR.

Reply:
Sounds good